src/TraceEvent/TraceUtilities/PEFile.cs

```diff
@@ -351,32 +352,52 @@ internal void FreeBuff(PEBuffer buffer)
     sealed unsafe class PEHeader
     {
         /// <summary>
-        /// Returns a PEHeader for void* pointer in memory.  It does NO validity checking. 
+        /// Returns a PEHeader that references an existing buffer without copying. Validates buffer bounds.
         /// </summary>
-        public PEHeader(void* startOfPEFile)
+        internal PEHeader(PEBufferedSlice slice)
         {
-            dosHeader = (IMAGE_DOS_HEADER*)startOfPEFile;
-            if (dosHeader->e_magic != IMAGE_DOS_HEADER.IMAGE_DOS_SIGNATURE)
+            m_slice = slice;
+            
+            var span = slice.AsSpan();
+            if (span.Length < sizeof(IMAGE_DOS_HEADER))
             {
                 goto ThrowBadHeader;
             }
 
-            var imageHeaderOffset = dosHeader->e_lfanew;
+            IMAGE_DOS_HEADER dosHdr = MemoryMarshal.Read<IMAGE_DOS_HEADER>(span);
+            
+            if (dosHdr.e_magic != IMAGE_DOS_HEADER.IMAGE_DOS_SIGNATURE)
+            {
+                goto ThrowBadHeader;
+            }
+
+            var imageHeaderOffset = dosHdr.e_lfanew;
             if (!(sizeof(IMAGE_DOS_HEADER) <= imageHeaderOffset && imageHeaderOffset <= 512))
```

Comment:
Please remove the imageHeaderOffset <= 512 requirement.  Specifically, we do want to make sure that the offset is at least sizeof(IMAGE_DOS_HEADER), but otherwise, it can be larger than 512.